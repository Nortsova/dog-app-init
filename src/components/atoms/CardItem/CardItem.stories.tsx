import type { Meta, StoryObj } from '@storybook/react';
import { CardItem } from '.';

const meta = {
  title: 'Atoms/CardItem',
  component: CardItem,
  argTypes: {
    variant: {
      options: ['primary', 'secondary'],
      control: 'radio'
    }
  }
} satisfies Meta<typeof CardItem>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    variant: 'primary'
  }
};

export const Secondary: Story = {
  args: {
    variant: 'secondary'
  }
};
