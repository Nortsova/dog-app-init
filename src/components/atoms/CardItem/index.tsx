export const CardItem = ({ variant }: any) => <div>Card Item {variant}</div>;
