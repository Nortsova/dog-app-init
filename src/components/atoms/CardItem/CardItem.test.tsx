import { composeStories } from '@storybook/react';
import * as stories from './CardItem.stories';
import { testSnapshot } from '../../../test/utils/snapshot.utils';

describe('CardItem', () => {
  const { Primary, SmallRadiusBorder, BigRadiusBorder } =
    composeStories(stories);
  testSnapshot(<Primary />);
  testSnapshot(<SmallRadiusBorder />);
  testSnapshot(<BigRadiusBorder />);
});
