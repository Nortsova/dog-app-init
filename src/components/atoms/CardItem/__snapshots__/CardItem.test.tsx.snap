// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardItem should match snapshot Component 1`] = `
<div>
  <div>
    Card Item 
    primary
  </div>
</div>
`;

exports[`CardItem should match snapshot Component 2`] = `
<div>
  <div>
    Card Item 
    secondary
  </div>
</div>
`;
